scripts/tunnel-creation.mjs

```diff
@@ -0,0 +1,348 @@
+  const tlsOptions = {
+    rejectUnauthorized: false,
+    minVersion: 'TLSv1.2',
+  };
+
+  log.info('Connecting to usbmuxd...');
+  const usbmux = await createUsbmux();
+
+  try {
+    log.info('Listing all connected devices...');
+    const devices = await usbmux.listDevices();
+
+    if (devices.length === 0) {
+      log.warn('No devices found. Make sure iOS devices are connected and trusted.');
+      return;
+    }
+
+    log.info(`Found ${devices.length} connected device(s):`);
+    devices.forEach((device, index) => {
+      log.info(`  ${index + 1}. UDID: ${device.Properties.SerialNumber}`);
+      log.info(`     Device ID: ${device.DeviceID}`);
+      log.info(`     Connection: ${device.Properties.ConnectionType}`);
+      log.info(`     Product ID: ${device.Properties.ProductID}`);
+    });
+
+    let devicesToProcess = devices;
+    if (specificUdid) {
+      devicesToProcess = devices.filter(
+        (device) => device.Properties.SerialNumber === specificUdid,
+      );
+
+      if (devicesToProcess.length === 0) {
+        log.error(`Device with UDID ${specificUdid} not found in connected devices.`);
+        log.error('Available devices:');
+        devices.forEach((device) => {
+          log.error(`  - ${device.Properties.SerialNumber}`);
+        });
+        process.exit(1);
+      }
+    }
+
+    log.info(`\nProcessing ${devicesToProcess.length} device(s)...`);
+
+    /** @type {import('appium-ios-remotexpc').TunnelResult[]} */
+    const results = [];
+
+    for (const device of devicesToProcess) {
+      const result = await tunnelCreator.createTunnelForDevice(device, tlsOptions);
+      results.push(result);
+    }
+
+    log.info('\n=== TUNNEL CREATION SUMMARY ===');
+    const successful = results.filter((r) => r.success);
+    const failed = results.filter((r) => !r.success);
+
+    log.info(`Total devices processed: ${results.length}`);
+    log.info(`Successful tunnels: ${successful.length}`);
+    log.info(`Failed tunnels: ${failed.length}`);
+
+    if (successful.length > 0) {
+      log.info('\n✅ Successful tunnels:');
+      const registry = await tunnelCreator.updateTunnelRegistry(results);
+      await startTunnelRegistryServer(registry, tunnelCreator._tunnelRegistryPort);
+
+      log.info('\n📁 Tunnel registry API:');
+      log.info('   The tunnel registry is now available through the API at:');
+      log.info(`   http://localhost:${tunnelCreator._tunnelRegistryPort}/remotexpc/tunnels`);
+      log.info('\n   Available endpoints:');
+      log.info('   - GET /remotexpc/tunnels - List all tunnels');
+      log.info('   - GET /remotexpc/tunnels/:udid - Get tunnel by UDID');
+      log.info('   - GET /remotexpc/tunnels/metadata - Get registry metadata');
+      if (successful.length > 0) {
+        const firstUdid = successful[0].device.Properties.SerialNumber;
+        log.info(`   curl http://localhost:4723/remotexpc/tunnels/${firstUdid}`);
+      }
+    }
+  } finally {
```

Comment:
perhaps it would make sense to extract the content of the try/catch block to a separate method to improve readability and simplify the main method